### ISSUE TITLE
Safe-AA Error: My tx only get succeed if I increase safeTxGas and BaseGas manually in code. otherwise it gets fails

### ISSUE THREAD

**Issue by sunnyRK** [safe-global/safe-core-sdk#521]
If I relayTransaction like below it gets fails
```
const dumpSafeTransafer: any[] = [
      {
          to: "0x2791Bca1f2de4661ED88A30C99A7a9449Aa84174",
          data: approveTx.data,
          value: ethers.utils.parseUnits("0").toString(),
          operation: OperationType.Call,
      },
  ];
```
                
 and If i add transaction with manually like safeTxGas and baseGas it gets succeed
 
 ```
const dumpSafeTransafer: any[] = [
  {
      to: "0x2791Bca1f2de4661ED88A30C99A7a9449Aa84174",
      data: approveTx.data,
      value: ethers.utils.parseUnits("0").toString(),
      operation: OperationType.Call,
      safeTxGas: BigNumber.from("45568").toNumber(),
      baseGas: BigNumber.from("124615538075886000")
  },
];
```
 
 
 How to decide programatically safeTxGas and baseGas.

**Comment by dasanra** [comment]
Hi @sunnyRK 

we are about to release a version of the SDK where you shouldn't need to specify gas parameters in order to get a successful transaction. Although I would like to get some details that could help us to understand your specific case

Which SDK version are you using?
Which kit are you using?
Which SDK method are you calling with that data?
Which network were you trying to execute?

Thank you!

**Comment by sunnyRK** [comment]
Hey @dasanra 

I am using auth-kit and relay-kit 
Initially i was using latest sdk vesions but it was failing and then i shift to below,
So now, I am using these sdk versions as metioned in account-abstraction-demo-ui repo,
   ```
 "@safe-global/account-abstraction-kit-poc": "^1.0.1",
    "@safe-global/auth-kit": "^1.2.0",
    "@safe-global/onramp-kit": "^1.3.0",
    "@safe-global/relay-kit": "^1.0.0",
    "@safe-global/safe-react-components": "^2.0.3",
    "@web3auth/base": "^4.6.0",
    "@web3auth/modal": "^4.6.2",
    "@web3auth/openlogin-adapter": "^4.6.0",
```

I am calling relayTransaction method to execute like below,
                `const gelatoTaskId = await safeAccountAbstraction.relayTransaction(dumpSafeTransafer, options);`

**Comment by sunnyRK** [comment]
Hey any update on above

**Comment by dasanra** [comment]
Hey @sunnyRK we published a new version of the relay-kit. 

To sync all we will need you to set
```
"@safe-global/account-abstraction-kit-poc":  ^1.3.0
"@safe-global/relay-kit":  ^1.3.0
```

Currently the `account-abstraction-kit` as its name suggest it's a PoC so you can sort this using directly that package, with less intervention or directly using the relay-kit. With `account-abstraction-kit` it will be possible to use Safes having only one signature required.

If you try to send a transaction paying with balance of a native token that you hold in your Safe I recommend you tweak the transaction object like this:

```
const dumpSafeTransafer: any[] = [
      {
          to: "0x2791Bca1f2de4661ED88A30C99A7a9449Aa84174",
          data: approveTx.data,
          value: ethers.utils.parseUnits("0").toString()
      },
  ];
```

And then call it like this:
```
const gelatoTaskId = await safeAccountAbstraction.relayTransaction(dumpSafeTransafer)
```

If you try to use 1Balance or using an ERC20 that you hold in your Safe you will need to tweak the options a bit:
```
  // options for using 1Balance
  const options: MetaTransactionOptions = {
    isSponsored: true
  }
```

```
  // options for using an ERC20
  const options: MetaTransactionOptions = {
    gasToken: GAS_TOKEN_ADDRESS
  }
```

You can find compatible GAS_TOKENS_ADDRESS for each chain
https://docs.gelato.network/developer-services/relay/payment-and-fees/syncfee-payment-tokens

Please test this and hope all works as expected now.

**Comment by dasanra** [comment]
This issue should be completely solved in `relay-kit v2.0.0` as there is no need to set the gas for relaying any more.
If you still find issues please feel free to open a new ticket.
